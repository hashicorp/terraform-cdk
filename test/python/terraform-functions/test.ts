// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import { TestDriver } from "../../test-helper";

describe("python terraform functions test synth", () => {
  let driver: TestDriver;

  beforeAll(async () => {
    driver = new TestDriver(__dirname);
    await driver.setupPythonProject();
  });

  test("synth generates JSON", async () => {
    await driver.synth();
    expect(
      driver.synthesizedStack("python-simple").toString()
    ).toMatchSnapshot();
  });
});
