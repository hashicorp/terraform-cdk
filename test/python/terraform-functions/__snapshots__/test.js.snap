// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`python terraform functions test synth synth generates JSON 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"python-simple\\",
      \\"backend\\": \\"local\\",
      \\"overrides\\": {
        \\"stack\\": [
          \\"terraform\\"
        ],
        \\"null_resource\\": [
          \\"triggers\\"
        ]
      }
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"null\\": {
        \\"version\\": \\"~> 3.1.0\\",
        \\"source\\": \\"hashicorp/null\\"
      }
    },
    \\"backend\\": {
      \\"remote\\": {
        \\"organization\\": \\"test\\",
        \\"workspaces\\": {
          \\"name\\": \\"test\\"
        }
      }
    }
  },
  \\"provider\\": {
    \\"null\\": [
      {}
    ]
  },
  \\"resource\\": {
    \\"null_resource\\": {
      \\"nullresource\\": {
        \\"triggers\\": {
          \\"cluster_instance_ids\\": \\"foo\\"
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"python-simple/null-resource\\",
            \\"uniqueId\\": \\"nullresource\\"
          }
        }
      }
    }
  },
  \\"output\\": {
    \\"computed\\": {
      \\"value\\": \\"\${element(merge({id = null_resource.nullresource.id}, {value = \\\\\\"123\\\\\\"}), 1)}\\"
    }
  }
}"
`;
